quint/src/builtin.qnt

```diff
@@ -964,8 +964,52 @@ module builtin {
   // /// pure val s = tuples(Set(1, 2), Set(3, 4))
   // /// assert(s == Set((1, 3), (1, 4), (2, 3), (2, 4)))
   // /// ```
-
   // pure def tuples(a*): (Set[a0], ..., Set[an]) => Set[(a0, ..., an)]
+
+  // /// `variant("Label", expr)` is a value of type `T` when
+  // ///
+  // /// - `T` is a predifined sum type
+  // /// - `T` has a data constructor `Label`, paired with type `E`
+  // /// - `expr` has type `E`
+  // ///
+  // /// ### Examples
+  // ///
+  // /// ```
+  // /// type T = A(int) | B | C(str)
+  // /// assert(variant("A", 42) == A(42))
+  // /// assert(variant("B", Rec()) == B)
+  // /// assert(variant("C", "foo") == C("foo"))
+  // /// ```
+  // ///
+  // /// Note that this is an internal built-in and users should prefer the generated
+  // /// data constructor matching the label of a variant. E.g., in our example above,
+  // /// users should prefer the equivalent form on the right hand side of the `==`.
+  // def variant(label, x): (str, E) => T
+
```

Comment:
This is an idea for the future. Shall we add a prefix to the names of the internal operators, e.g., `__variant` or `q::variant`. This would send the users a clear message that these operators should be used with care.

Reply:
Not a bad idea, but it will make the normalized pretty printing harder to read I think unless we have custom printers for all the hidden operators?